netplan/cli/commands/apply.py

```diff
@@ -60,6 +60,12 @@ def run(self):  # pragma: nocover (covered in autopkgtest)
         self.parse_args()
         self.run_command()
 
+    @staticmethod
+    def get_alt_names(iface):
+        ret = subprocess.run(['ip', 'link', 'show', iface], capture_output=True)
+        data = ret.stdout.decode('utf-8').split()
+        return [data[i + 1] for i, x in enumerate(data) if x == "altname"]
```

Comment:
Maybe we could call `['ip', '-j', 'link', 'show', iface]` instead and then use something like this to parse/access the data, which should be more robust than enumerating over the .split() chunks:

```
data = yaml.safe_load(ret.stdout.decode('utf-8'))
if data and len(data) == 1:
    return data[0].get('altnames', [])
return []
```

Reply:
Yes, that seems like a good idea. I did not even know about the -j option.